compiler/visq/visqlib/QErrorComputer.py

```diff
@@ -83,8 +99,9 @@ class MPEIRComputer(QErrorComputer):
     def __init__(self, fp32_dir, fq_dir):
         super().__init__(fp32_dir, fq_dir)
 
-    def run(self):
-        qerror_map = dict()
+    # Incrementally compute Qerror while traversing all data in fp32_dir and fq_dir
+    def advance_on(self, fp32_dir, fq_dir):
+        self.update_dirs(fp32_dir, fq_dir)
```

Comment:
Hmm.. I think there are some redundancies that make it hard to read code.

1.. Directory names are used in `advance_on` and abandoned. So setting `self._fp32_dir` and `self._fq_dir` in `update_dirs` seems unnecessary. It would be better to pass `fp32_dir` and `fq_dir` to `update_data`, and remove `update_dirs`.

2. After 1 is applied, can you return a `data_path` (dict) in `update_data` rather than saving it to `self._data_path` ? It seems that `self._data_path` also does not have to be saved.

Reply:
ok.

Reply:
done